improver_tests/wxcode/wxcode/test_ModalFromGroupings.py

```diff
@@ -0,0 +1,509 @@
+    if reverse_wet_values:
+        wet_categories = {}
+        for key in WET_CATEGORIES.keys():
+            wet_categories[key] = [i for i in reversed(WET_CATEGORIES[key])]
+    if reverse_wet_keys:
+        wet_categories = dict(reversed(list(wet_categories.items())))
+
+    result = ModalFromGroupings(
+        BROAD_CATEGORIES, wet_categories, INTENSITY_CATEGORIES, wet_bias=wet_bias,
+    )(wxcode_cubes)
+    expected = [expected] if not isinstance(expected, list) else expected
+    for index in range(len(expected)):
+        assert result.data.flatten()[index] == expected[index]
+
+
+@pytest.mark.parametrize("record_run_attr", [False])
+@pytest.mark.parametrize("model_id_attr", [False, True])
+@pytest.mark.parametrize("interval", [1])
+@pytest.mark.parametrize("offset_reference_times", [False, True])
+@pytest.mark.parametrize("cube_type", ["gridded", "spot"])
+@pytest.mark.parametrize(
+    "data, day_weighting, day_start, day_end, day_length, expected",
+    (
+        # All weather codes supplied are considered as daytime.
+        # There are more light shower codes, so this is the modal code.
+        ([1, 1, 1, 1, 10, 10, 10, 10, 10], 1, 0, 9, 24, 10),
+        # For a day length of 9 and a day weighting of 2, the number of clear day codes
+        # doubles with one more shower symbol giving 6 dry codes, and 5 wet codes.
+        ([1, 1, 1, 1, 1, 10, 10, 10, 10], 2, 5, 7, 9, 1),
+        # Altering the day_end to 16Z results in 6 dry codes in total and 6 wet codes,
+        # so the resulting code is wet.
+        ([1, 1, 1, 1, 10, 10, 10, 10, 10], 2, 4, 7, 9, 10),
+        # A day weighting of 2 results in the number of clear day codes doubling,
+        # and one more shower symbol giving 6 dry codes, and 5 wet codes.
+        ([1, 1, 1, 1, 1, 10, 10, 10, 10], 2, 5, 7, 24, 1),
+        # Altering the day_end to 16Z results in 6 dry codes in total and 6 wet codes,
+        # so the resulting code is wet.
+        ([1, 1, 1, 1, 10, 10, 10, 10, 10], 2, 4, 7, 24, 10),
+        # Increasing the day weighting to 3 results in 8 dry codes and 7 wet codes, so
+        # the resulting code is dry.
+        ([1, 1, 1, 1, 10, 10, 10, 10, 10], 3, 4, 7, 9, 1),
+        # An example for two points for the first point being dry, and the second point
+        # being wetter, with day weighting resulting in a dry modal code.
+        (
+            [[1, 1, 1, 1, 1, 1, 1, 1, 1], [1, 1, 1, 1, 10, 10, 10, 10, 10]],
+            3,
+            4,
+            7,
+            9,
+            [1, 1],
+        ),
+        # An example for two points for the first point being mostly dry, and the
+        # second point being wetter, with day weighting resulting in a dry modal code.
+        (
+            [[1, 1, 1, 1, 10, 1, 1, 1, 1], [1, 1, 1, 1, 10, 10, 10, 1, 1]],
+            3,
+            2,
+            5,
+            9,
+            [1, 10],
+        ),
+    ),
+)
+def test_expected_values_day_weighting(
```

Comment:
@gavinevans and I have had a chat and concluded that some of the comments associated with this test are not consistent with what they are describing.